python/pyproject.toml

```diff
@@ -18,7 +18,7 @@ classifiers = [
     "Programming Language :: Python :: 3.12"
 ]
 dependencies = [
-    "pyarrow>=8"
+    "pyarrow>=14.0.1"
```

Comment:
We shouldn't do this. It will essentially make it impossible to build a venv with a lot of libraries that use for some reason the older pyarrow version.

Azure has this issue particularly..

Reply:
Thanks @ion-elgreco  - I reverted this back to `pyarrow>=8` and added pyarrow-hotfix as a dependency.  Note, there are no conflicts when installing `pyarrow==14.0.1` and `pyarrow-hotfix`.

Reply:
Shouldn't it also be imported, otherwise the hotfix doesn't do anything I presume

Reply:
Ugh - my bad!  Working on this!

Reply:
@ion-elgreco What do you think of this approach?  Thanks!

Reply:
Are these pyarrow_hotfix stubs needed?

Reply:
No I just meant the stub file, creating an empty __init__.pyi was sufficient